src/core/gui.py

```diff
@@ -0,0 +1,229 @@
+    :type time_interval: int
+    :param get_interval_emissions_method: method to compute the equivalent carbon emission
+    :type get_interval_emissions_method: function
+    :param num_x_points: number of points to be plotted in the x-axis
+    :type num_x_points: int
+    :param num_x_ticks: number of ticks to be set in the x-axis
+    :type num_x_ticks: int
+    :param x_unit_measurement: letter corresponding to the unit of measurement for the x-axis (s: seconds, m: minutes, h:hours)
+    :type x_unit_measurement: str
+    :param y_unit_measurement: letter corresponding to the unit of measurement for the y-axis (s: seconds, m: minutes, h:hours)
+    :type y_unit_measurement: str
+    
+    '''
+    def __init__(self, cpu_tdp, co2_intensity, time_interval, get_interval_emissions_method, num_x_points, num_x_ticks, x_unit_measurement, y_unit_measurement):
+        super().__init__()
+
+        # initialize attributes
+        self.get_interval_emissions_method = get_interval_emissions_method
+        self.time_interval = time_interval
+        self.co2_intesity = co2_intensity
+        self.cpu_tdp = cpu_tdp
+        self.num_x_points = num_x_points
+        self.num_x_ticks = num_x_ticks
+        self.x_unit_measurement = x_unit_measurement
+        self.y_unit_measurement = y_unit_measurement
+        self.thread_running = False
+        self.new_value = 0
+
+        # find y unit of measurement multiplier
+        if y_unit_measurement == "s":
+            self.y_unit_multiplier = 1
+        elif y_unit_measurement == "m":
+            self.y_unit_multiplier = 60
+        elif y_unit_measurement == "h":
+            self.y_unit_multiplier = 3600
+        else:
+            raise ValueError("Please enter a valid unit of measure for the y-axis. Choose between 's', 'm' or 'h'.")
+
+        # initialize initial graphs values
+        self.x = np.arange(num_x_points)[::-1]*-1
+        self.y = self.x*0
+
+        # call initialization methods
+        self.create_window()
+        self.insert_ax()
+        self.connect_timer()
+
+    def create_window(self):
+        '''
+        This method creates a new window layout and prepare a white canvas where a plot can be drawn.
+        '''
+        self.setWindowTitle('Energy Monitor Plot')
+        layout = QVBoxLayout()
+        self.setLayout(layout)
+        fig = plt.Figure(figsize=(15, 3))
+        self.canvas = FigureCanvas(fig)
+        layout.addWidget(self.canvas)
+
+    def insert_ax(self):
+        '''
+        This method populate the empty canvas with a matplotlib axes structure.
+        '''
+        # define matplotlib font properties
+        font = {
+            'weight': 'normal',
+            'size': 10
+        }
+        matplotlib.rc('font', **font)
+
+        # create a matplotlib line plot structure
+        self.ax = self.canvas.figure.subplots()
+        self.ax.set_xlim([-self.num_x_points, 0])
```

Comment:
this is not working properly, leaving one space more than expected. not a big deal but can be an easy fix

Reply:
nice catch, thanks